frontend/src/api/actions/applications.js

```diff
@@ -25,6 +30,23 @@ const deleteOneApplicationSuccess = actionFactory(
     DELETE_ONE_APPLICATION_SUCCESS
 );
 
+const applicantToApplicantId = flattenIdFactory("applicant", "applicant_id");
+const positionToPositionId = flattenIdFactory("position", "position_id");
+function prepForApi(data) {
+    const propName = "position_preferences";
+    const [ret, filtered] = splitObjByProps(data, [propName]);
+
+    if (filtered.length <= 0) {
+        return data;
```

Comment:
This doesn't make sense to me...we should always have an `applicant` that needs to be flattened.

Reply:
Good point, I'll fix this

Reply:
So, I changed the if condition to be the opposite: `if (filtered.length > 0)`. Put the mapping of `position_preference` inside the `if` scope so that